packages/generator/ni_measurement_plugin_sdk_generator/client/__init__.py

```diff
@@ -60,89 +62,109 @@ def _create_file(
     "--class-name",
     help="Name for the Python Measurement Plug-In Client Class in the generated module.",
 )
+@click.option(
+    "-a",
+    "--all",
+    is_flag=True,
+    help="Creates Python Measurement Plug-In Client for all the active measurement services.",
+)
 @click.option(
     "-o",
     "--directory-out",
     help="Output directory for Measurement Plug-In Client files. Default: '<current_directory>/<module_name>'",
 )
 def create_client(
-    measurement_service_class: str,
+    measurement_service_class: List[str],
     module_name: Optional[str],
     class_name: Optional[str],
+    all: Optional[bool],
     directory_out: Optional[str],
 ) -> None:
     """Generates a Python Measurement Plug-In Client module for the measurement service.
 
     You can use the generated module to interact with the corresponding measurement service.
 
     MEASUREMENT_SERVICE_CLASS: The service class of the measurement.
+    Provide space separated service class to generate multiple clients.
     """
     channel_pool = GrpcChannelPool()
     discovery_client = DiscoveryClient(grpc_channel_pool=channel_pool)
     built_in_import_modules: List[str] = []
     custom_import_modules: List[str] = []
 
-    if module_name is None or class_name is None:
-        base_service_class = measurement_service_class.split(".")[-1]
-        base_service_class = remove_suffix(base_service_class)
-        if not base_service_class.isidentifier():
+    if all:
+        measurement_service_class = get_service_class_list(discovery_client)
+    else:
+        if not measurement_service_class:
+            raise click.ClickException("Measurement service class cannot be empty.")
+
+    for service_class in measurement_service_class:
+        if can_generate_name(module_name, class_name, measurement_service_class):
+            base_service_class = service_class.split(".")[-1]
+            base_service_class = remove_suffix(base_service_class)
+            if not base_service_class.isidentifier():
+                raise click.ClickException(
+                    "Unable to create client.\nPlease provide a valid module name or update the measurement with a valid service class."
+                )
+
+            if module_name is None or len(measurement_service_class) > 1:
```

Comment:
If the user wants to create clients for more than one measurement, we won't use the user-provided class or module names. Instead, we will generate our own module and class names.

Reply:
`len(measurement_service_class) > 1` checks for multiple module generation. I have updated this by giving a name.